File: 1528-shuffle-string/1528-shuffle-string.js
/**
 * @param {string} s
 * @param {number[]} indices
 * @return {string}
 */
var restoreString = function(s, indices) {
    let ret = [];
    for (let i = 0; i < indices.length; ++i) {
        ret[indices[i]] = s.charAt(i);
    }
    return ret.join('');
};

File: 1528-shuffle-string/1528-shuffle-string.cpp
class Solution {
public:
    string restoreString(string s, vector<int>& indices) {
        string res;
        res.resize(s.size());
        for(int i = 0; i < indices.size(); i++)
        {
            res[indices[i]] = s[i];
        }
        return res;
    }
};